perception/sub8_batcave_ogrid/src/OGridGen.cpp

```diff
@@ -0,0 +1,142 @@
+  pubPointCloud = nh.advertise<sensor_msgs::PointCloud2>("/point_cloud", 1);
+
+  nh.param<float>("resolution", resolution, 0.2f);
+  nh.param<float>("ogrid_size", ogrid_size, 50.f);
+  nh.param<float>("pool_depth", pool_depth, 7.f);
+  
+  int pointCloudBufferSize;
+  nh.param<int>("buffer_size", pointCloudBufferSize, 5000);
+  pointCloudBuffer.resize(pointCloudBufferSize);
+
+  get_bounds = nh.serviceClient<sub8_msgs::Bounds>("get_bounds");
+
+  timer = nh.createTimer(ros::Duration(0.3), std::bind(&OGridGen::publish_ogrid, this, std::placeholders::_1));
+  sub = nh.subscribe("/blueview_driver/ranges", 1, &OGridGen::callback, this);
+}
+
+void OGridGen::publish_ogrid(const ros::TimerEvent &)
+{
+  // Draw ogrid
+  sub8_msgs::Bounds bound_hold;
+  if (get_bounds.call(bound_hold))
+  {
+    bounds = std::vector<cv::Point>(4);
+    size_t k = 0;
+    for (auto &p : bound_hold.response.bounds)
+    {
+      bounds.at(k) = (cv::Point(p.x / resolution + matOgrid.cols / 2, p.y / resolution + matOgrid.rows / 2));
+      k++;
+    }
+    const cv::Point *pts = (const cv::Point *)cv::Mat(bounds).data;
+    int npts = cv::Mat(bounds).rows;
+    cv::polylines(matOgrid, &pts, &npts, 1, true, 255, 3, CV_8U, 0);
+  }
+
+  // Flatten Ogrid
+  nav_msgs::OccupancyGrid rosGrid;
+  std::vector<int8_t> data(matOgrid.cols*matOgrid.rows);
+  size_t k = 0;
+  for (int i = 0; i < matOgrid.cols; ++i)
+  {
+    for (int j = 0; j < matOgrid.rows; ++j)
+    {
+      data.at(k) = ((int)matOgrid.at<uint8_t>(cv::Point(i, j)));
+      k++;
+    }
+  }
+
+  // Publish Ogrid
+  rosGrid.header.seq = 0;
+  rosGrid.info.resolution = resolution;
+  rosGrid.header.frame_id = "map";
+  rosGrid.header.stamp = ros::Time::now();
+  rosGrid.info.map_load_time = ros::Time::now();
+  rosGrid.info.width = matOgrid.cols;
+  rosGrid.info.height = matOgrid.rows;
+  rosGrid.info.origin.position.x = -ogrid_size / 2;
+  rosGrid.info.origin.position.y = -ogrid_size / 2;
+  rosGrid.info.origin.position.z = 0;
+  rosGrid.data = data;
+  pubGrid.publish(rosGrid);
+
+  // Publish point cloud
+  pcl::PointCloud<pcl::PointXYZI> pointCloud;
+  pointCloud.resize(pointCloudBuffer.capacity());
+  size_t l = 0;
+  for (auto &p : pointCloudBuffer)
+  {
+    pointCloud.at(l) = p;
+    l++;
+  }
+  sensor_msgs::PointCloud2 output;
+  pcl::toROSMsg(pointCloud, output);
+  output.header.frame_id = "map";
+  output.header.stamp = ros::Time::now();
+  pubPointCloud.publish(output);
+
+  // if(pointCloud.size() > 4000)
+  // classification.findNormals(pointCloud);
+}
+
+void OGridGen::callback(const blueview_driver::BlueViewPingPtr &ping_msg)
+{
+  tf::StampedTransform transform;
+  try
+  {
+    listener.lookupTransform("/map", "/base_link", ros::Time(0), transform);
+  }
+  catch (tf::TransformException ex)
+  {
+    ROS_DEBUG_STREAM("Did not get TF for sonar");
+    return;
+  }
+
+  matOgrid = cv::Mat::zeros((int)(ogrid_size / resolution), (int)(ogrid_size / resolution), CV_8U);
```

Comment:
You could minimize the use of parenthesis here and elsewhere: `int(ogrid_size / resolution)`